### PR TITLE
Feat:  Docs lineage for change categories

### PR DESCRIPTION
**Pull request by mykalmax**
show column level lineage on column click in docs
Screenshot 2023-05-05 at 4 13 21 PM

show lineage in change category
Screenshot 2023-05-05 at 4 14 11 PM
a264-fe4180123296.jpg)

### CODE REVIEW

File: web/client/src/library/pages/docs/SourceList.tsx [hunk 1]
```diff
@@ -52,23 +51,35 @@ export default function SourceList({
...
-                  className={clsx(
-                    'py-1 px-4 rounded-md w-full hover:bg-primary-10',
-                    activeModel === model.name
+              <NavLink
+                to={`${EnumRoutes.IdeDocsModels}/${ModelSQLMeshModel.encodeName(
+                  model.name,
+                )}`}
+                className={({ isActive }) =>
+                  clsx(
+                    'block px-2 overflow-hidden whitespace-nowrap overflow-ellipsis py-1  rounded-md w-full hover:bg-primary-10',
```
Comment (vchan) [inline web/client/src/library/pages/docs/SourceList.tsx:62]:
```suggestion
                    'block px-2 overflow-hidden whitespace-nowrap overflow-ellipsis py-1 rounded-md w-full hover:bg-primary-10',
```

File: web/client/src/models/sqlmesh-model.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+          continue
+        }
+      }
+    }
+
+    return lineage
+  }
+
+  static encodeName(modelName: string): string {
```
Comment (vchan) [inline web/client/src/models/sqlmesh-model.ts:151]: Why is this needed?

File: web/client/src/models/sqlmesh-model.ts
Comment (vchan) [inline web/client/src/models/sqlmesh-model.ts:1]: What's the naming convention for file names?

**Review by vchan — APPROVED** [review]

File: web/client/src/models/sqlmesh-model.ts (hunk 2 above)
Comment (mykalmax) [reply web/client/src/models/sqlmesh-model.ts:151]: if we do `/docs/models/sushi.customers` browser thinks `.customers` is top-level domain i guess, so i'm juts replacing `.`

File: web/client/src/library/components/graph/Graph.tsx [hunk 3]
```diff
@@ -395,67 +461,71 @@ function ModelColumn({
...
+  )
+}
+
+ModelColumn.Display = function Display({
+  columnName,
+  columnType,
+  columnDescription,
+  isFetching = false,
+  isError = false,
+  disabled = false,
```
[the remark this answers is not in the document]
Comment (mykalmax) [reply web/client/src/library/components/graph/Graph.tsx:485]: `disabled` potentially can be attribute on html element  , same with `selected`. So in components `disabled` prop is juts `disabled`

File: web/client/src/models/sqlmesh-model.ts
Comment (mykalmax) [reply web/client/src/models/sqlmesh-model.ts:1]: react has it's own naming convention for files. 
but models here are kinda outside the framework , so it is usually kebab-case
